Initial setup

itsjeffro opened the pull request:
**Added**

- readme
- github action (tests for php 7.3, 7.4 and does a composer update for lowest and highest dependency version)
- separated api classes based on their categories, eg the issueRefund and getOrder APIs belong to the ebay fulfillment api group. This will make it easier to maintain if we end up adding more apis for the public to use in this package.
- added tests

**Note**

- The composer dependencies are currently based on the one's in xpay. when we update the composer dependencies in xpay, the we can go ahead and update the dependencies in this sdk.

Commit:
Added tests setup

Commit:
Added .env example and gitignore

Commit:
Added initial api class

Currently with fulfillment apis getOrder and issueRefund

Commit:
Point to example env

Commit:
Updated test namespace, added gitkeep

Commit:
Added vcr for tests

Commit:
Added composer require step

Commit:
Copy .env

Commit:
Use yaml for vcr tapes

Commit:
Update request bodies

Commit:
Removed cover report

Commit:
Bumped minor version from 6.0 to 6.5 for guzzlehttp

Commit:
Updated steps

Commit:
Remove redundant config pararm from constructor

Commit:
Updated with badges

specialtactics commented on composer.json:
```diff
@@ -0,0 +1,39 @@
+    },
+    "require": {
+        "php": "^7.3",
+        "guzzlehttp/guzzle": "^6.5",
```
Hey Jeff, laravel 8 will require guzzle ^7.0.1, can you please make that the minunm? (Laravel 7 also supports this version).

itsjeffro replied:
@specialtactics 

I changed this to "guzzlehttp/guzzle": "^6.5|^7.1"

It looks like xpay has some other dependencies that still require `guzzlehttp`. Eg. `prometheus-push-gateway-php-client`

So i'll make a task to get those to a newer `guzzlehttp` version so that xpay isn't bogged down with old versions.

specialtactics replied:
Yep, makes sense.

Commit:
Bumped guzzlehttp version from 6 to 7, phpunit from 9 to 9.3

This is so we can eventually support laravel 8 when we upgrade from laravel 7

Commit:
Allow support for guzzlehttp 6.5 support

We still have some other packages being by xpay api that require guzzlehttp ^6.0
So we'll have to slowly update those packages to a newer version of guzzle. Eg,
prometheus-push-gateway-php-client

Commit:
Changed back to 1.1 to avoid conflicts with guzzle http 6.5

mith495 commented on src/AbstractClient.php:
```diff
@@ -0,0 +1,49 @@
+
+        parent::__construct(
+            $client,
+            $this->buildDescription("$baseUri"),
```
$baseUri meant to be a variable instead of a string inside the double quote here?

itsjeffro replied:
@mith495 No, it's correct. 

$baseUri returns an instance of a class that has a `__toString()` magic method. So when the $baseUri is treated as a string, __toString will connect all the related uri segments and return a string.

itsjeffro replied:
@mith495 Just for reference. This is the magic method Guzzle uses in its Uri class ...

```
    public function __toString()
    {
        return self::composeComponents(
            $this->scheme,
            $this->getAuthority(),
            $this->path,
            $this->query,
            $this->fragment
        );
    }
```
